foreshadow/transformers/transformers.py

```diff
@@ -58,187 +92,288 @@ def wrap_transformer(transformer):
     Returns:
         The wrapped form of a transformer
 
-    """
-    members = [
-        m[1]
-        for m in inspect.getmembers(transformer, predicate=inspect.isfunction)
-    ]
-    wrap_candidates = [
-        m
-        for m in members
-        if any(
-            name == m.__name__
-            for name in [
-                "fit",
-                "fit_transform",
-                "transform",
-                "inverse_transform",
-            ]
-        )
-    ]
-
-    for w in wrap_candidates:
-        # Wrap public function calls
-        method = pandas_partial(pandas_partial(w))
-        method.__doc__ = w.__doc__
-        setattr(transformer, w.__name__, method)
-
-    # Wrap constructor
-    if "__defaults__" in dir(transformer.__init__):
-        setattr(
-            transformer,
-            "__init__",
-            Sigcopy(transformer.__init__)(init_partial(transformer.__init__)),
-        )
-    else:
-        setattr(transformer, "__init__", init_replace)
-
-    return transformer
-
-
-class Sigcopy(object):
-    """Copy the argspec between two functions.
-
-    Used to copy the argspec from a partial function.
-
-    Params:
-        src_func: The source function to copy
-
-    Call Params:
-        tgt_func: The target function to emulate
+    ..#noqa: I401
 
     """
-
-    def __init__(self, src_func):
-        """Save necessary info to copy over.
-
-        Args:
-            src_func: The source function to be copied.
-
-        """
-        self.argspec = inspect.getfullargspec(src_func)
-        self.src_doc = src_func.__doc__
-        self.src_defaults = src_func.__defaults__
-
-    def __call__(self, tgt_func):
-        """Run when Sigcopy object is called. Returns new function.
-
-        Args:
-            tgt_func: The target function to emulate
-
-        Returns:
-            source function wrapped as target function.
-            TODO(@Adithya) fix this docstring.
-
-        """
-        tgt_argspec = inspect.getfullargspec(tgt_func)
-
-        name = tgt_func.__name__
-
-        # Filters out defaults that are metaclasses
-        argspec = self.argspec
-        argspec = (
-            argspec[0:3]
-            + (
-                tuple(
-                    [
-                        s if type(s).__name__ != "type" else None
-                        for s in (argspec[3] if argspec[3] is not None else [])
-                    ]
-                ),
+    # use the same base
+    # metaclass as the transformer, otherwise we will get MRO metaclass
+    # issues in DFTransformer if we try to choose the base class for our
+    # metaclass that is not the same one for the transformer we are also
+    # extending.
+    class DFTransformerMeta(type(transformer)):
```

Comment:
Does the metaclass need to be defined inside the method?

Reply:
Yes, because the type of the Meta depends on the type of the transformer)

Reply:
Yes - needs to be based off the type of the transformer being in, which could, for example, be metaclassing type or abcmeta